### PR TITLE
Feature/store configs

### PR DESCRIPTION
**Pull request by it-sam**
PR: Add capability to store and retrieve man up config.

### CODE REVIEW

File: lib/src/ui/manup_widget.dart [hunk 1]
```diff
@@ -74,8 +74,8 @@ class _ManUpWidgetState extends State<ManUpWidget>
 //
   @override
   void dispose() {
-    widget.service.close();
-    WidgetsBinding.instance?.removeObserver(this);
     super.dispose();
+    widget?.service?.close();
+    WidgetsBinding.instance?.removeObserver(this);
```
Comment (SilkyPantsDan) [inline lib/src/ui/manup_widget.dart:79]: Might want to remove the observer before `super.dispose`?

File: pubspec.yaml [hunk 2]
```diff
@@ -14,6 +14,7 @@ dependencies:
   http: ^0.12.0
   meta: ^1.1.8
   url_launcher: ^5.4.2
+  path_provider: any
```
Comment (SilkyPantsDan) [inline pubspec.yaml:17]: Would prefer this to have a version range - `^1.0.0` maybe (so anything < v2.0)

File: lib/src/manup_service.dart [hunk 3]
```diff
@@ -116,9 +125,29 @@ class ManUpService {
...
+    }
+  }
+
+  Future<Metadata> _readManupFile() async {
+    try {
+      var data = await fileStorage.readfile();
+      Map<String, dynamic> json = jsonDecode(data);
+      return Metadata(data: json);
+    } catch (e) {
+      throw e;
```
Comment (zbarbuto) [inline lib/src/manup_service.dart:144]: having this in a try-catch is pointless if all you do is re-throw the error.

**Comment by zbarbuto** [comment]
I'm confused as to why you'd want to store the manup config? Isn't the whole point that it refreshes on every app load?

**Comment by SilkyPantsDan** [comment]
> I'm confused as to why you'd want to store the manup config? Isn't the whole point that it refreshes on every app load?

If the app is taken 'Offline' after getting a lockout, it could then sidestep the whole ManUp process

**Comment by zbarbuto** [comment]
Ah I see - so effectively caching the last killswitch etc. Makes sense.

**Pull request merged**